File: dot-net/src/OCP.Filter/OCP.Filter/Model/Product.cs
namespace OCP.Filter.Model
{
    public class Product
    {
        public Product(ProductColor color)
        {
            Color = color;
        }

        public Product(ProductColor color, ProductSize size)
        {
            Color = color;
            Size = size;
        }

        public ProductColor Color { get; set; }
        public ProductSize Size { get; set; } 
    }
}

File: java/ocp-filter/src/Product.java

public class Product {
	private final ProductColor color;
	private final ProductSize size;
	
    public Product(ProductColor color) {
        this(color, ProductSize.Medium);
    }

    public Product(ProductColor color, ProductSize size) {
        this.color = color;
        this.size = size;
    }

    public ProductColor getColor() {
    	return this.color;
    }
    
    public ProductSize getSize() {
    	return this.size;
    } 
}